### ISSUE TITLE
Binderhub refusing to launch server

### ISSUE THREAD

**Issue by sandertyu** [LibreTexts/metalc#238]
binder.libretexts.org is currently refusing to launch the jupyter hub server for all repositories. The error on binderhub just looks like this;

![binderhub-error](https://cdn.discordapp.com/attachments/763198235237154837/858572799518179338/unknown.png)

I suspect it could be something with some secret or api token, and/or something to do with our recent k8 upgrades in general. There's no logs that I have found which point to anything more enlightening. This issue is preventing CKEditor-binder-plugin from working at all as well.

**Comment by sandertyu** [comment]
We believe this is the main error, seen in the logs of the `binder` pod;

```
[W 210627 05:58:25 health:209] Unhealthy services: [{'service': 'JupyterHub API', 'ok': False}]
[E 210627 05:58:25 log:140] 503 GET /health (anonymous@172.27.188.113) 751.70ms
[I 210627 05:58:30 builder:534] Launching pod for https://github.com/LibreTexts/default-env: 0 other pods running this repo (0 total)
[I 210627 05:58:30 launcher:166] Creating user libretexts-default-env-8cvnilxe for image libretexts/binder-dev-libretexts-2ddefault-2denv-1cb626:d5bc2100e26493399e69972452ed1a0df9743cf3
[E 210627 05:58:30 builder:593] Retrying launch of https://github.com/LibreTexts/default-env after error (duration=0s, attempt=1): ConnectionRefusedError(111, 'Connection refused')
[I 210627 05:58:34 launcher:166] Creating user libretexts-default-env-9q6pr7p4 for image libretexts/binder-dev-libretexts-2ddefault-2denv-1cb626:d5bc2100e26493399e69972452ed1a0df9743cf3
[E 210627 05:58:34 builder:593] Retrying launch of https://github.com/LibreTexts/default-env after error (duration=0s, attempt=2): ConnectionRefusedError(111, 'Connection refused')
[I 210627 05:58:40 health:116] Getting pod statistics for binderhub
[E 210627 05:58:40 health:51] Error checking check_jupyterhub_api
    Traceback (most recent call last):
      File "/usr/local/lib/python3.8/site-packages/binderhub/health.py", line 49, in wrapper
        res = await f(*args, **kwargs)
      File "/usr/local/lib/python3.8/site-packages/binderhub/health.py", line 27, in wrapper
        return await f(*args, **kwargs)
      File "/usr/local/lib/python3.8/site-packages/binderhub/health.py", line 141, in check_jupyterhub_api
        await AsyncHTTPClient().fetch(hub_url + "hub/health", request_timeout=3)
    ConnectionRefusedError: [Errno 111] Connection refused
```

There's probably some sort of token that we don't have set properly.

**Comment by moorepants** [comment]
Ask in the jupyter forums or chat.

**Comment by sandertyu** [comment]
> Ask in the jupyter forums or chat.

We did, but we've found the issue and we're pretty sure its HAproxy which is not configured correctly on the second router. We've bumped the first router to be master for now and we'll fix HAproxy on the second.

**Comment by TimStewartJ** [comment]
We've fixed the HAProxy issue on fw2, but this has brought our attention to a bigger carp issue that has yet to be resolved.

**Comment by sandertyu** [comment]
We believe we have fixed HAproxy, and that the issue was with CARP. On fw2, HAproxy was misconfigured, so we tried to make fw1 our master while we could work on it. This caused an incomplete CARP master/backup transfer (reason unknown) and the internal interface of fw2 remained as master while all others were backup. This caused the network to fully go down. 

What we have done is made fw2 the master again (haproxy is fixed and working) and brought down the cxgbe0 interface on fw1. All services are entirely operational again, but we'll want to debug CARP.
